torch/_higher_order_ops/associative_scan.py

```diff
@@ -63,8 +92,15 @@ def associative_scan(
             This function must be pure, pointwise, and satisfy the associative property.
         input (torch.Tensor): The input tensor, or nested pytree of tensors.
             All inputs are expected to have the same shape.
-        dim (int): the dimension to scan over
-
+        dim (int): The dimension to scan over
+        reverse (bool): A boolean stating if the scan should be reversed with respect to the dimension.
+        generic_scan (bool): A boolean stating whether a generic scan mode should be used. 
+            If the generic scan mode is ``not used``, there are restrictions on the operations allowed
+            within the ``combine_fn``. For example, only pointwise functions are currently allowed.
+            If the generic scan mode is ``used``, the restrictions on the combine_fn are less strict, 
+            for example also non-pointwise operations are allowed. 
+            However, the non-generic scan mode utilizes efficient ``tl.associative_scan`` calls and is thus
+            more efficient as the generic scan mode.
```

Comment:
A suggestion for how to write it a bit more concisely.
```suggestion
        If ``generic_scan=False``, only pointwise ops may be used within ``combine_op``. ``generic_scan=False``
        may just be used on CUDA tensors. On the other hand, ``generic_scan=False`` should be more
        efficient than ``generic_scan=True`` whenever it can be used.
```

Reply:
Done